### PR TITLE
Button & Icon button updated based on new Design Template

### PR DESCRIPTION
**Pull request by JulienTexier**
## ✨ What has changed?

Button & Icon button updated based on new Design Template

## 🔍 How to verify?

https://github.com/TaitoUnited/react-native-template/assets/45203192/20a715d9-c779-43c9-8825-50ae7475ac91

The playground shows all the possible variations of the buttons.

### CODE REVIEW

File: src/app/(auth)/landing.tsx
Comment (Temzasse) [inline src/app/(auth)/landing.tsx]: Should the `forcedColor` prop be removed?

File: src/app/playground/_layout.tsx
Comment (Temzasse) [inline src/app/playground/_layout.tsx]: Remove console log.

File: src/components/uikit/buttons/Button.tsx
Comment (Temzasse) [inline src/components/uikit/buttons/Button.tsx]: In fullstack template we decided to allow only one icon and have a separate prop for the icon position. Maybe we should do the same here?

File: src/components/uikit/buttons/Button.tsx
Comment (Temzasse) [inline src/components/uikit/buttons/Button.tsx]: Should the `9` be `0.9` or something else?

File: src/components/uikit/buttons/Button.tsx
Comment (Temzasse) [inline src/components/uikit/buttons/Button.tsx:28]: Could you use variants and compoundVariants for these dynamic styles?

File: src/components/uikit/buttons/IconButton.tsx
Comment (Temzasse) [inline src/components/uikit/buttons/IconButton.tsx]: Maybe instead we give the prop a default value?

File: src/app/(auth)/landing.tsx
Comment (JulienTexier) [reply src/app/(auth)/landing.tsx]: `neutral` gives a black color (`text`), while here we need it white. But I agree it is not ideal to have, maybe we should simply remove this landing icon from using the uikit `IconButton`

File: src/components/uikit/buttons/Button.tsx
Comment (JulienTexier) [reply src/components/uikit/buttons/Button.tsx]: The `.` magically disappeared from my attention 🙂‍↔️

File: src/components/uikit/buttons/Button.tsx
Comment (JulienTexier) [reply src/components/uikit/buttons/Button.tsx:28]: No mention at all about `compoundVariants` in the [library's documentation](https://github.com/Temzasse/stitches-native)... I need to have a word or two with its maintainer 😋

File: src/app/(auth)/landing.tsx
Comment (JulienTexier) [reply src/app/(auth)/landing.tsx]: Black is not so bad after all in the landing page, so I removed `forcedColor` entirely

File: src/components/uikit/buttons/Button.tsx
Comment (JulienTexier) [reply src/components/uikit/buttons/Button.tsx:28]: Until they are fixed in stitches, we are staying with the dynamic styles

**Comment by JulienTexier** [comment]
The Button section of the playground now has the loading and disabled state

https://github.com/TaitoUnited/react-native-template/assets/45203192/07767bd2-2d62-4afa-ba6a-6dbe08fba197

**Review by Temzasse — APPROVED** [review]

**Pull request merged**
